dDatabase/CDClientDatabase/CDClientTables/CDPhysicsComponentTable.cpp

```diff
@@ -4,32 +4,31 @@ void CDPhysicsComponentTable::LoadValuesFromDatabase() {
 	auto tableData = CDClientDatabase::ExecuteQuery("SELECT * FROM PhysicsComponent");
 	auto& entries = GetEntriesMutable();
 	while (!tableData.eof()) {
-		CDPhysicsComponent entry;
-		entry.id = tableData.getIntField("id", -1);
+		const uint32_t componentID = tableData.getIntField("id", -1);
+
+		auto& entry = entries[componentID];
+		entry.id = componentID;
 		entry.bStatic = tableData.getIntField("static", -1) != 0;
 		entry.physicsAsset = tableData.getStringField("physics_asset", "");
-		UNUSED(entry->jump = tableData.getIntField("jump", -1) != 0);
-		UNUSED(entry->doublejump = tableData.getIntField("doublejump", -1) != 0);
-		entry.speed = tableData.getFloatField("speed", -1);
-		UNUSED(entry->rotSpeed = tableData.getFloatField("rotSpeed", -1));
-		entry.playerHeight = tableData.getFloatField("playerHeight");
-		entry.playerRadius = tableData.getFloatField("playerRadius");
+		UNUSED_COLUMN(entry.jump = tableData.getIntField("jump", -1) != 0;)
+		UNUSED_COLUMN(entry.doubleJump = tableData.getIntField("doublejump", -1) != 0;)
+		entry.speed = static_cast<float>(tableData.getFloatField("speed", -1));
+		UNUSED_COLUMN(entry.rotSpeed = tableData.getFloatField("rotSpeed", -1);)
+		entry.playerHeight = static_cast<float>(tableData.getFloatField("playerHeight"));
+		entry.playerRadius = static_cast<float>(tableData.getFloatField("playerRadius"));
```

Comment:
These are already floats in the entry struct, not sure why we're casting it to the type that is already returned here

Reply:
and that is of course because getFloatField does not return a float but a double.  Very cool api